--- ganeti-instance-image-0.5.1.orig/common.sh	2012-12-16 04:55:48.259404952 +0000
+++ ganeti-instance-image-0.5.1.new/common.sh	2012-12-16 04:55:22.809667928 +0000
@@ -24,6 +24,7 @@
 SFDISK="/sbin/sfdisk"
 QEMU_IMG="/usr/bin/qemu-img"
 MKDIR_P="/bin//mkdir -p"
+INSTANCE_MEM=
 
 CLEANUP=( )
 
@@ -159,6 +160,10 @@
 
 format_disk0() {
     local sfdisk_cmd="$SFDISK -uM -H 255 -S 63 --quiet --Linux --DOS $1"
+    if [ "${SWAP}" = "yes" ] && [ -z "$SWAP_SIZE" ] ; then
+        log_error "SWAP_SIZE not set however SWAP is enabled"
+        exit 1
+    fi
     if [  "${SWAP}" = "yes" -a -z "${KERNEL_PATH}" ] ; then
         # Create three partitions:
         # 1 - 100MB /boot, bootable
@@ -364,14 +369,16 @@
     . "$DEFAULT_FILE"
 fi
 
-# note: we don't set a default mirror since debian and ubuntu have
-# different defaults, and it's better to use the default
+# Variable name change from 2.5 to 2.6
+if [ -n "$INSTANCE_BE_memory" ] ; then
+    INSTANCE_MEM="$INSTANCE_BE_memory"
+else
+    INSTANCE_MEM="$INSTANCE_BE_maxmem"
+fi
 
-# only if the user want to specify a mirror in the defaults file we
-# will use it, this declaration is to make sure the variable is set
 : ${CDINSTALL:="no"}
 : ${SWAP:="yes"}
-: ${SWAP_SIZE:="${INSTANCE_BE_memory}"}
+: ${SWAP_SIZE:="${INSTANCE_MEM}"}
 : ${FILESYSTEM:="ext3"}
 : ${KERNEL_ARGS=""}
 : ${OVERLAY=""}
